mne/cov.py

```diff
@@ -307,6 +307,54 @@ def make_ad_hoc_cov(info, verbose=None):
     return Covariance(data, ch_names, info['bads'], info['projs'], nfree=0)
 
 
+##############################################################################
+# Create from array
+# TODO: add test for this
+@verbose
+def make_custom_cov(info, standard_deviations, verbose=None):
+    """Create an ad hoc noise covariance.
+
+    Parameters
+    ----------
+    info : instance of Info
+        Measurement info.
+    standard_deviations : dict of string : (numpy) array
+        stds of the diagonal elements
+        keys must be eeg, grad and mag
```

Comment:
we can enhance `make_ad_hoc_cov` to allow a `std` dict rather than adding a new function

Reply:
Agreed, it will avoid all the code repetitions. I just didn't wanna mess up with your code before reviews.

Reply:
Yep, and a tip -- make the default `None` and use `std = _handle_default(std)` to populate the dict (hopefully it's clear, if not, see how we do this elsewhere)